### PR TITLE
Integrate Nightwatch into GitHub Actions

### PR DESCRIPTION
**Pull request by U-DON**
## Description
Related to department-of-veterans-affairs/va.gov-team#25311.

## Testing done
CI passes.

## Acceptance criteria
- [ ] Nightwatch tests should be included in the GitHub Actions CI workflow.

## Definition of done
- [ ] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [ ] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

File: .github/workflows/continuous-integration.yml [hunk 1]
```diff
@@ -243,11 +243,81 @@ jobs:
...
 
+      - name: Generate build details
+        run: |
+          cat > build/${{ matrix.buildtype }}/BUILD.txt << EOF
+          BUILDTYPE=${{ matrix.buildtype }}
+          BRANCH_NAME=$(echo "${GITHUB_REF#refs/heads/}")
+          RUN_ID=${{ github.run_id }}
+          REF=${{ github.sha }}
+          BUILDTIME=$(date +%s)
+          EOF
```
Comment (U-DON) [inline .github/workflows/continuous-integration.yml:254]: Moved the generation of BUILD.txt into the build step. That way, the archive step won't have to do an extra set of decompression and compression.

Even if the step were to only add the file to the existing tar file, I figured having the artifacts be consistent with the S3 uploaded files (by having both with the BUILD.txt) would be better and maybe reduce confusion down the line.

Also felt right to create `BUILD.txt` at the time of the actual build.

File: .github/workflows/continuous-integration.yml [hunk 2]
```diff
@@ -243,11 +243,81 @@ jobs:
...
+
+      - name: Compress and archive build
+        run: tar -C build/${{ matrix.buildtype }} -cjf ${{ matrix.buildtype }}.tar.bz2 .
```
Comment (U-DON) [inline .github/workflows/continuous-integration.yml:257]: Significantly reduced sizes help speed up the artifact uploads and downloads.

File: .github/workflows/continuous-integration.yml [hunk 3]
```diff
@@ -243,11 +243,81 @@ jobs:
...
+          name: ${{ matrix.buildtype }}.tar.bz2
+          path: ${{ matrix.buildtype }}.tar.bz2
+
+  nightwatch:
+    name: Nightwatch E2E Tests
+
+    runs-on: self-hosted
+
+    env:
+      NODE_EXTRA_CA_CERTS: /etc/ssl/certs/VA-Internal-S2-RCA1-v1.cer.pem
```
Comment (U-DON) [inline .github/workflows/continuous-integration.yml:271]: Prefer setting this only in self-hosted runner jobs in order to reduce noise in errors in the GitHub runners about not being able to find the certs.

**Review by oseasmoran73 — APPROVED** [review]
LGTM!

File: .github/workflows/continuous-integration.yml [hunk 4]
```diff
@@ -243,11 +243,77 @@ jobs:
...
+            node_modules
+          key: ${{ steps.get-node-version.outputs.NODE_VERSION }}-${{ hashFiles('yarn.lock') }}
+
+      - name: Install dependencies
+        run: yarn install --frozen-lockfile --prefer-offline
+        env:
+          YARN_CACHE_FOLDER: ~/.cache/yarn
+
+      - name: Run Nightwatch tests
+        run: yarn test:e2e:headless
```
Comment (ddzz) [inline .github/workflows/continuous-integration.yml:313]: Would it make sense to add a timeout here, or are Nightwatch tests stable enough?

Comment (U-DON) [reply .github/workflows/continuous-integration.yml:313]: They've been stable from what I've seen, but certainly doesn't hurt to add a timeout for the job.

**Pull request merged**